src/modules/auth/token.service.ts

```diff
@@ -81,10 +79,8 @@ export class TokenService {
     }
 
     if (token) {
-      const decodedToken = this.jwtService.decode(token) as TokenPayload;
-      if (decodedToken.supportedOrigins.includes(requestOrigin) ||
-        decodedToken.supportedHosts.includes(req.hostname) // swagger and server based requests check only
-      ) {
+      const decodedToken = service.decode(token) as TokenPayload;
+      if (decodedToken.origin === req.headers['origin'] || req.headers['origin'] === undefined) {
```

Comment:
Would be good to explain why the logic is as it is
```suggestion
      const isBrowserRequestFromAuthenticatedOrigin = decodedToken.origin === req.headers['origin'] 
      const isServerRequestOrGETFromSameDomain =  req.headers['origin'] === undefined
      if (isBrowserRequestFromAuthenticatedOrigin || isServerRequestOrGETFromSameDomain) {
```